### PR TITLE
pythonPackages.flask-autoindex: init at 0.6

### PR DESCRIPTION
**Pull request by timokau**
###### Motivation for this change

~~Depends on #38783, do not merge before that.~~

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/flask-autoindex/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  propagatedBuildInputs = [
+    flask
+    flask-silk
+    future
+  ];
+
+  meta = with stdenv.lib; {
+    description = "The mod_autoindex for Flask";
+    longDescription = ''
+      Flask-AutoIndex generates an index page for your Flask application automatically. The result just like mod_autoindex, but the look is more awesome!
```
[the remark this answers is not in the document]
Comment (timokau) [reply pkgs/development/python-modules/flask-autoindex/default.nix]: Is there a standard for `longDescription` line length? In my opinion non-semantic linebreaks should be done by whatever renders the description.

[the remark this answers is not in the document]
Comment (dotlambda) [reply pkgs/development/python-modules/flask-autoindex/default.nix]: I don't know of any. But if there's a `.` in your `longDescription`, you should definitely add a line break.

File: pkgs/development/python-modules/flask-autoindex/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    flask
+    flask-silk
+    future
+  ];
+
+  meta = with stdenv.lib; {
+    description = "The mod_autoindex for Flask";
+    longDescription = ''
+      Flask-AutoIndex generates an index page for your Flask application automatically.
+      The result just like mod_autoindex, but the look is more awesome!
```
Comment (dotlambda) [inline pkgs/development/python-modules/flask-autoindex/default.nix]: I think there's a verb missing

**Comment by timokau** [comment]
I rebased on master since the dependency is now merged.

**Comment by dotlambda** [comment]
@GrahamcOfBorg build python2.pkgs.flask-autoindex python3.pkgs.flask-autoindex

File: pkgs/development/python-modules/flask-autoindex/default.nix [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  propagatedBuildInputs = [
+    flask
+    flask-silk
+    future
+  ];
+
+  postInstall = ''
+    # binary does not run, known issue and fixed in next release (>0.6)
+    # see https://github.com/sublee/flask-autoindex/pull/32
+    rm $out/bin/fai
```
Comment (dotlambda) [inline pkgs/development/python-modules/flask-autoindex/default.nix]: I don't think there will be a next release. Maybe use https://github.com/sublee/flask-autoindex/pull/32.patch as a patch?

Comment (timokau) [reply pkgs/development/python-modules/flask-autoindex/default.nix]: You might be right. Done.

**Comment by dotlambda** [comment]
@GrahamcOfBorg build python2.pkgs.flask-autoindex python3.pkgs.flask-autoindex

**Pull request merged**